wiki/src/main/resources/XWiki/PDFViewerMacro.xml

```diff
@@ -172,6 +172,14 @@
       <code>{{include reference="Licenses.Code.VelocityMacros"/}}
 
 {{velocity output="false"}}
+#macro (checkDocumentAsAuthor $documentName)
+  #if (!$services.security.authorization.hasAccess('view', $documentName) &amp;&amp; $asAuthor &amp;&amp;
+      $xwiki.hasProgrammingRights() &amp;&amp; $xwiki.exists('XWiki.PDFViewerService'))
+    #set ($document = $xwiki.getDocumentAsAuthor($documentName))
+    #set ($authorRightsNeeded = true)
+  #end
```

Comment:
I would:

* rename the macro to ``maybeGetDocumentAsAuthor($documentReference $asAuthor)``
* add an ``else`` here to retrieve the document based on the given reference or full name
* call the ``#maybeGetDocumentAsAuthor`` macro below passing either the document full name or its reference, removing the preceding ``#set ($document = ...)`` line.

WDYT?